frame/support/src/lib.rs

```diff
@@ -444,6 +444,21 @@ macro_rules! parameter_types_impl_thread_local {
 					pub fn set(t: $type) {
 						[<$name:snake:upper>].with(|v| *v.borrow_mut() = t);
 					}
+
+					/// Mutate the internal value in place.
+					pub fn mutate<R, F: FnOnce(&mut $type) -> R>(mutate: F) -> R{
+						let mut current = Self::get();
+						let result = mutate(&mut current);
+						Self::set(current);
+						result
+					}
+
+					/// Get and remove the value.
+					pub fn take() -> $type {
+						let current = Self::get();
+						Self::set(Default::default());
```

Comment:
Hmm isn't this weird?

Maybe setting it to `$value`, but still weird :see_no_evil:

Reply:
I'm not so used to macros, why is it weird? And how can I improve it?

Reply:
By weird I mean that it is reset to `Default`.  You specify the "default" value and if you take this, I would expect that you get this value again.

Reply:
The default value isn't being specified here though.

Reply:
Line 428 you got `$value`. That is the default I'm speaking about.

Reply:
Oh, the value got at 458 is different from the default.

Reply:
I think I understand now. Sorry about the confusion, not too literate with macro syntax.